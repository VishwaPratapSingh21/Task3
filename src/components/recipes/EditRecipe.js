import React, { Component } from 'react';

/**
 * Component for editing recipes.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class EditRecipe extends Component{
    state ={
      title:'',
      description:''
    }
    handleInput =(event) =>{
      const {name, value} = event.target;
      this.setState({[name]:value})
    }
    handleEditRecipe =(event) =>{
      event.preventDefault();
      this.props.editRecipe(this.props.id,this.state.title,this.state.description)
    }
    componentWillMount(){
      this.setState({title:this.props.title, description:this.props.description})
  }
    render(){
      return(
        <div
            className="modal fade"
            id={`edit_recipe${this.props.id}`}
            role="dialog"
            aria-labelledby="exampleModalLabel"
            aria-hidden="true">
            <div className="modal-dialog" role="document">
                <div className="modal-content">
                    <div className="modal-header">
                        <h5 className="modal-title">Edit a recipe</h5>
                        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                        </button>
                    </div>
                    <form onSubmit={this.handleEditRecipe}>
                    <div className="modal-body">
                    {this.props.error ? (
                    <div className="alert alert-danger">{this.props.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
                        <div className="form-group">
                            <input type="text" name="title" onChange={this.handleInput} className='form-control' value={this.state.title}/>
                        </div>
                        <div className="form-group">
                                <textarea
                                    rows="4"
                                    cols="50"
                                    name="description"
                                    className="form-control description"
                                    placeholder="description or instructions"
                                    onChange={this.handleInput}
                                    value={this.state.description}
                                  >
                                  </textarea>
                              </div>
                    </div>
                    <div className="modal-footer">
                        <button type="submit" className="btn btn-primary" >Update</button>
                        <button type="button" className="btn btn-secondary" data-dismiss="modal" id ={`close${this.props.id}`}>Cancel</button>
                    </div>
                    </form>
                </div>
            </div>
        </div>
      );
    }
  }
  export default EditRecipe;