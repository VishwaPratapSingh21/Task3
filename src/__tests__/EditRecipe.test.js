import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import EditRecipe from '../components/recipes/EditRecipe';

describe('EditRecipe component', () => {
  const wrapper = shallow(<EditRecipe  editRecipe={jest.fn()}/>);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has divs in a component', () => {
    expect(wrapper.find('div')).toHaveLength(8);
  });

  it('has headers h5 in a component', () => {
    expect(wrapper.find('h5')).toHaveLength(1);
  });
  it('has a form in a modal', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });

  it('has span in a modal', () => {
    expect(wrapper.find('span')).toHaveLength(1);
  });

  it('has button in a modal', () => {
    expect(wrapper.find('button')).toHaveLength(3);
  });

  it('has inputs in a modal', () => {
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('form').simulate('submit', {preventDefault}))
    expect(preventDefault).toBeCalled();
    expect(wrapper.find('[name="title"]').simulate('change', {target:{name:'title', value:'hi'}}))
  });

  it('calls component did mount', () => {
    sinon.spy(EditRecipe.prototype, 'componentWillMount');
    const fullWrapper = mount(<EditRecipe />);
    expect(EditRecipe.prototype.componentWillMount).toHaveProperty('callCount', 1);
    EditRecipe.prototype.componentWillMount.restore();
  });
});
