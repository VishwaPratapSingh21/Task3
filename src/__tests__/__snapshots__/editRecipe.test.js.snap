// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipe component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="edit_recipeundefined"
  role="dialog"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Edit a recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="title"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="form-control description"
              cols="50"
              name="description"
              onChange={[Function]}
              placeholder="description or instructions"
              rows="4"
            />
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Update
          </button>
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            id="closeundefined"
            type="button"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
